### ISSUE TITLE
XML Image Drag and Drop Issue

### ISSUE THREAD

**Issue by maryfries** [jmoenig/Snap#3204]
We are so exited about these XML images!! 

We have lots to do to figure out how we'll implement them, but in the meantime, Michael and I caught this glitch when we were playing around dragging XML images from BJC (local copy) into Snap. It seems like if we don't click in Snap after dragging one in, even though it appears to drop, when we drag in the next one, we'll still have the previous one attached to the mouse pointer.

It would be ideal to get that fixed. Any ideas? Thanks! :)

**Comment by maryfries** [comment]
https://user-images.githubusercontent.com/6549787/229154169-d150e0a5-14b9-4a58-82ee-79f82ae9e800.mov

**Comment by jguille2** [comment]
Mary, I can't reproduce this.

Which browser are you using? Or maybe it's an effect of your desktop configuration, or your video capture system?

When I do this... After dragging (when release the button) the image doesn't drop anywhere... it is just attached to the pointer... and so, you need to click to release the pointer.
That behavior (the image dropping down just after releasing the dragging action) is not what I see in my tests.

**Comment by maryfries** [comment]
Hi Joan! I don't know which computer and browser I was on that day. On this machine (OSX 10.15.7, Chrome 111.0.5563.146, using my local copy of BJC and my local copy of Snap synced to 34eb21ff477af2f7afc5c2848fab65c9a3324cec; I'm using a local copy for CORS reasons), if I drag XML images from BJC into Snap, I get XML-free costumes, and if I download them first and them drag them in, I get the actual blocks _without_ the aforementioned issue.

**Comment by maryfries** [comment]
...and I can still replicate it on the same machine in Safari version 15.6.1 (15613.3.9.1.16, 15613). I don't think it's my video capture system as I'm not using it at all now.

**Comment by maryfries** [comment]
Actually, I think I remember Jens saying that Safari is weird. Are there browser requirements listed anywhere on the Snap site? BJC currently says "For the best experience with Snap!, make sure your browser is up to date." Should that be changed to say that Chrome is preferred over Safari (and perhaps any other browsers)?

**Comment by brianharvey** [comment]
I don't think that page should discuss specific browsers unless someone's going to keep it up to date forever.  The preferred browser keeps changing as one after another introduce new bugs.

**Comment by jguille2** [comment]
Hi Mary
I see some "browser issues" dragging images... but I'm a little confused about "this issue".

- First. I was not able to reproduce at any time what you showed in the video. Dragging issues yes, but not that behavior.
- Second. I was specially confused, because I was trying to test with the same page you were reporting... but I didn't see that those images had xml added (just current pictures). But finally I guess you are testing with another branch of the repo (not the main one...I guess is the "csp" branch)... and then, I've found those enriched pictures.

And what I see is:
- Dragging pictures (xml enriched or just plain pictures) has issues. We have CORS... but also we have "browser issues". In my test (on Linux) Chromium is doing it well... but Firefox doesn't.
- But when "drag and drop" is running, I don't see issues and XMLenriched pictures are doing what I expect from them.

I guess if your current issue (not the original) is about cache... And maybe your browser is serving you the cached "simple image" if you navigated before on the main branch of the repo copy.

Joan

**Comment by cycomachead** [comment]
Oh! I think this might be a https / http issue, but I'll need to test some more.
